lib/experimental/fonts/font-library/class-wp-rest-font-collections-controller.php

```diff
@@ -69,24 +69,27 @@ public function register_routes() {
 	 * @param WP_REST_Request $request Full details about the request.
 	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
 	 */
-	public function get_font_collection( $request ) {
+	public function get_item( $request ) {
 		$slug       = $request->get_param( 'slug' );
 		$collection = WP_Font_Library::get_font_collection( $slug );
+
 		// If the collection doesn't exist returns a 404.
 		if ( is_wp_error( $collection ) ) {
 			$collection->add_data( array( 'status' => 404 ) );
-			return $collection;
+			return rest_ensure_response( $collection );
```

Comment:
I don't think you need to use `rest_ensure_response` when returning an error. All the controllers I've seen return the error directly.